### PR TITLE
x-pack/filebeat/input/internal: split httpmon out of httplog

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

This moves the metric monitoring code out into its own package as metrics aren't logs.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: x-pack/filebeat/input/internal/httpmon/roundtripper.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+	reqStart := time.Now()
+	resp, err := rt.transport.RoundTrip(req)
+	rt.metrics.roundTripTime.Update(time.Since(reqStart).Nanoseconds())
+
+	if resp != nil {
+		rt.metrics.resps.Add(1)
+	}
+
+	if resp == nil || err != nil {
```
Comment (ShourieG) [inline x-pack/filebeat/input/internal/httpmon/roundtripper.go:147]: any specific reason why we are not checking this immediately after the error is returned ?

Comment (efd6) [reply x-pack/filebeat/input/internal/httpmon/roundtripper.go:147]: This is all existing code with no semantic change. It was added in #35392.

**Review by ShourieG — APPROVED** [review]
LGTM

**Review by belimawr — APPROVED** [review]

**Comment by belimawr** [comment]
@efd6 could you also add an entry in the `CHANGELOG-developer.next.asciidoc`?

**Pull request merged**